frontend/lib/src/components/elements/DeckGlJsonChart/DeckGlJsonChart.tsx

```diff
@@ -144,35 +150,51 @@ export class DeckGlJsonChart extends PureComponent<PropsWithHeight, State> {
     return null
   }
 
-  static getDeckObject = (props: PropsWithHeight): DeckObject => {
+  static getDeckObject = (
+    props: PropsWithHeight,
+    state: Partial<State>
+  ): DeckObject => {
     const { element, width, height, theme } = props
-    const json = JSON.parse(element.json)
+    const { elementId } = state
+
+    const isFullScreen = Boolean(height)
+
+    // Only parse JSON when not transitioning to/from fullscreen, the element id changes, or theme changes
```

Comment:
maybe i should remove this comment since it's not necessary?

Reply:
You could also add one more sentence here why we need to prepare with these changes